staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -541,6 +549,27 @@ func InClusterConfig() (*Config, error) {
 	}, nil
 }
 
+// InClusterNamespace return the default namespace to be used
+// for namespaced API operations in the pod
```

Comment:
document the possible return values:
* the `ErrNotInCluster` return value
* the possibility of returning no error and an empty namespace (or change behavior to always return `default` as a namespace and document that)